CHANGELOG.md

```diff
@@ -1,3 +1,177 @@
+# 0.0.124 - XXX, 2024 - "XXX"
+
+## API Updates
+ * `rust-bitcoin` has been updated to 0.32. The new `bitcoin-io` crate is now
+   used for all IO traits, irrespective of the features set on LDK crates.
+   LDK crates no longer automatically force features on dependent crates where
+   possible, allowing different `std`/`no-std` settings between LDK and
+   rust-bitcoin crates (e.g. to disable `std` on LDK to ensure system time is
+   not accessed while using `bitcoin-io`'s `std` feature). (#3063, #3239, #3249).
+ * A new `lightning_types` crate was added which contains various top-level
+   types. Using types from `lightning::ln::features` or
+   `Payment{Hash,Preimage,Secret}` from `lightning::ln` or
+   `lightning::ln::types` is now deprecated. The new crate is re-exported as
+   `lightning::types` (#3234, #3253).
+ * `lightning` now depends on `lightning-invoice`, rather than the other way
+   around. The `lightning_invoice::payment` module has moved to
+   `lightning::ln::bolt11_payment` and `lightning_invoice::utils` to
+   `lightning::ln::invoice_utils` (#3234).
+ * Event handlers may now return errors, causing most events to be replayed
+   immediately without blocking the background processor. See documentation on
+   individual `Event`s for more information on replay (#2995).
+ * `ChannelDetails::balance_msat` is deprecated in favor of
+   `ChainMonitor::get_claimable_balances` and the `Balance`, which now contains
+   substantially more details and more accurately calculates a node-wide
+   balance when `Balance::claimable_amount_satoshis` are summed (#3212, #3247).
+ * `ConfirmationTarget` has two new variants - a `MaximumFeeEstimate` which can
+   help to avoid spurious force-closes by ensuring we always accept feerates up
+   to this value from peers as sane and a `UrgentOnChainSweep`, replacing
+   `OnChainSweep` and only being used when the on-chain sweep is urgent (#3268).
+ * All `ChannelMonitor`s are no longer persisted after each block connection,
+   instead spreading them out over a handful of blocks to reduce load spikes.
+   Note that this will increase the incidence of `ChannelMonitor`s which have
+   different best blocks on startup, requiring some additional chain replay
+   (but only on some `ChannelMonitor`s) on startup for `Listen` users (#2966).
+ * A new format for Rapid Gossip Sync data is now supported which contains
+   additional node metadata and is more extensible (#3098).
+ * `ChannelManager::send_payment_with_route` is now deprecated in favor of the
+   much easier to use `Channelmanager::send_payment`. Those who wish to manually
+   select the route such payments go over should do so by matching the
+   `payment_id` in `Router::find_route_with_id` (#3224).
```

Comment:
The last sentence seems confusing. `Router::find_route_with_id` will give you a `Route`, but don't you still need to call `ChannelManager::send_payment_with_route` to use it?